go/libraries/doltcore/sqle/enginetest/dolt_transaction_queries.go

```diff
@@ -1998,15 +1998,15 @@ var DoltConstraintViolationTransactionTests = []queries.TransactionTest{
 					"To allow transactions to be committed with constraint violations from a merge or transaction sequencing set @@dolt_force_transaction_commit=1.\n" +
```

Comment:
I think we're missing a script test that exactly reproduces the original problem. Not completely clear on the exact setup, but something about setting `@foreign_key_checks` to 0 and creating two foreign keys that would have ended up with the same name before this change.

Reply:
Good call. I got too focused on cleaning up all those FK unit tests! I added a script test to trigger the same customer error; pretty easy to trigger once you turn off `@@foreign_key_checks`.